bin/build_update_subfolders.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+
+import send_email
+import synapseclient
+import synapseutils
+import sys
+
+from typing import List, Union
+
+
+def build_subfolder(
+    syn: synapseclient.Synapse,
+    folder_name: str,
+    parent_folder: Union[str, synapseclient.Entity],
+) -> synapseclient.Entity:
+    """
+    Builds a subfolder under the designated ``parent_folder``.
+
+    Arguments:
+        syn: A Synapse Python client instance
+        folder_name: The name of the subfolder to be created
+        parent_folder: A synapse Id or Entity of the parent folder under which the subfolder will live
+
+    Returns:
+        The created Synapse Folder entity
+    """
+
+    # Create Folder object
+    subfolder = synapseclient.Folder(name=folder_name, parent=parent_folder)
+    subfolder = syn.store(obj=subfolder)
+
+    return subfolder
+
+
+# TODO: https://sagebionetworks.jira.com/browse/IBCDPE-809
+# def update_subfolders():
+#     """"""
+
+
+def update_permissions(
+    syn: synapseclient.Synapse,
+    subfolder: Union[str, synapseclient.Entity],
+    project_folder_id: str,
+    principal_id: str = None,
+    access_type: List[str] = [],
+):
+    """
+    Updates the permissions (local share settings) of the given Folder/File to change access for the given principalId.
+    By default it will always revoke all access types for all challenge participants and the public.
+
+    Arguments:
+        syn: A Synapse Python client instance
+        subfolder: The Folder whose permissions will be updated
+        project_folder_id: The Project Synapse ID
+        principal_id: The synapse ID to change permissions for
+        access_type: Type of permission to be granted
+    """
+
+    # New ACL has all access types revoked for everyone except Project maintainers by default
+    all_participants = syn.restGET(f"/entity/{project_folder_id}/challenge").get(
+        "participantTeamId"
+    )
+    registered_users = "273948"
+    public = "273949"
+
+    for id in [all_participants, registered_users, public]:
+        syn.setPermissions(subfolder, principalId=id, accessType=[])
+
+    # Also update the access type for the designated principalId if there is one
+    if principal_id:
+        syn.setPermissions(subfolder, principalId=principal_id, accessType=access_type)
+
+
+def build_update_subfolders(
+    project_name: str,
+    submission_id: str,
+    build_or_update: str,
+    subfolders: List[str] = ["workflow_logs", "predictions"],
+    only_admins: str = "predictions",
+    parent_folder: str = "Logs",
+):
+    """
+    This function can either build/rebuild a set of log subfolders under
+    a participant folder, or update an existing participant folder with
+    new ancilliary files.
+
+    The current Challenge Folder structure is as follows:
+    >Parent-Folder/
```

Comment:
nit: The concept is conveyed by the format you gave, but it may look nicer to format this as if you ran `tree` on the directory structure:

```
Parent-Folder/
|--Level 1 Subfolder (Submitter-Folder)/
|  |--Level 2 Subfolder/
|  |  |-- ...
```